src/kafka-file/languageservice/services/common.ts

```diff
@@ -0,0 +1,21 @@
+import { SerializationSetting } from "../../../client/serialization";
```

Comment:
I would like to avoid having some files in services which doesn't provide a service. By service I mean completion, codelens, hover or your inline activation.

So at first please remove your common.ts, produce.ts and consumer.ts.

I will review your code to explain my idea.